design/service_test.go

```diff
@@ -111,3 +111,73 @@ func TestErrorExprValidate(t *testing.T) {
 		}
 	}
 }
+
+func TestFlowExpr_EvalName(t *testing.T) {
+	const tokenURL = "http://domain/token"
+	const refreshURL = "http://domain/refresh"
+
+	cases := map[string]struct {
+		tokenURL   string
+		refreshURL string
+		expected   string
+	}{
+		"tokenURL test":   {tokenURL: tokenURL, refreshURL: "", expected: fmt.Sprintf("flow with token URL %q", tokenURL)},
+		"refreshURL test": {tokenURL: "", refreshURL: refreshURL, expected: fmt.Sprintf("flow with refresh URL %q", refreshURL)},
+	}
+
+	for k, tc := range cases {
+		fe := &FlowExpr{TokenURL: tc.tokenURL, RefreshURL: tc.refreshURL}
+		if actual := fe.EvalName(); actual != tc.expected {
+			t.Errorf("%s: got %#v, expected %#v", k, actual, tc.expected)
+		}
+	}
+}
+
+func TestSchemeExpr_EvalName(t *testing.T) {
+	cases := map[string]struct {
+		kind     SchemeKind
+		expected string
+	}{
+		"OAuth2Kind":    {kind: OAuth2Kind, expected: "OAuth2Security"},
+		"BasicAuthKind": {kind: BasicAuthKind, expected: "BasicAuthSecurity"},
+		"APIKeyKind":    {kind: APIKeyKind, expected: "APIKeySecurity"},
+		"JWTKind":       {kind: JWTKind, expected: "JWTSecurity"},
+		"NoKind":        {kind: NoKind, expected: "This case is panic"},
+	}
+
+	// panic recover
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("should have panicked!")
+		}
+	}()
+
+	for k, tc := range cases {
+		se := &SchemeExpr{Kind: tc.kind}
```

Comment:
It would be good to count how many tests do not panic to make sure there were 4. Otherwise the test could succeed even if one of the first 4 tests panicked.

Reply:
You're right.  I'll fix it.